hash_practice/exercises.py

```diff
@@ -2,18 +2,40 @@
 def grouped_anagrams(strings):
     """ This method will return an array of arrays.
         Each subarray will have strings which are anagrams of each other
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(n)
+        Space Complexity: O(n)
     """
-    pass
+    anagrams_map = {}
+    for word in strings:
+        sorted_word = "".join(sorted(word))
+        if sorted_word not in anagrams_map:
+            anagrams_map[sorted_word] = [word]
+        else:
+            anagrams_map[sorted_word].append(word)
+    return list(anagrams_map.values())
+
 
 def top_k_frequent_elements(nums, k):
     """ This method will return the k most common elements
         In the case of a tie it will select the first occuring element.
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(n)
+        Space Complexity: O(n)
     """
-    pass
+    frequency_hash = {}
+    most_frequent_items = []
+    if len(nums) == 0:
+        return []
+    for item in nums:
+        if item in frequency_hash:
+            frequency_hash[item] += 1
+        else:
+            frequency_hash[item] = 1
+    
+    for i in range(k):
+        highest_frequency = max(frequency_hash, key = lambda num: frequency_hash[num])
+        most_frequent_items.append(highest_frequency) 
+        frequency_hash.pop(highest_frequency)
+    return most_frequent_items
 
 
 def valid_sudoku(table):

```

Comment:
👍 Nice work, just a note since the board is always 9x9 and never grows, the time/space complexity can be considered O(1).